### PR TITLE
Implement `Debug` and `PartialEq` for `CmdLine`

### LINKED ISSUE #98: cmdline derive Debug & PartialEq
This is needed for tests.

### PR DESCRIPTION
**Pull request by alexandruag**
Fixes #98

### CODE REVIEW

File: src/cmdline/mod.rs [hunk 1]
```diff
@@ -337,6 +337,18 @@ impl From<Cmdline> for Vec<u8> {
     }
 }
 
+impl fmt::Debug for Cmdline {
```
Comment (andreeaflorescu) [inline src/cmdline/mod.rs:340]: Why not use derive?

File: src/cmdline/mod.rs [hunk 2]
```diff
@@ -337,6 +337,18 @@ impl From<Cmdline> for Vec<u8> {
...
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.line)
+    }
+}
+
+impl PartialEq for Cmdline {
+    fn eq(&self, other: &Self) -> bool {
+        self.line == other.line
```
Comment (andreeaflorescu) [inline src/cmdline/mod.rs:348]: Why not check the capacity as well? In which case we could also just derive partialeq.

File: src/cmdline/mod.rs (hunk 1 above)
Comment (alexandruag) [reply src/cmdline/mod.rs]: I liked this format better, but I've added a derive now.

File: src/cmdline/mod.rs (hunk 2 above)
Comment (alexandruag) [reply src/cmdline/mod.rs:343]: It looked more natural to not involve the capacity in checks for equality (i.e. the same as happens for `Vec` IIRC). I can also add another derive if there's a strong preference for that.

File: src/cmdline/mod.rs [hunk 3]
```diff
@@ -479,4 +486,18 @@ mod tests {
         assert!(cl.insert_multiple("foo", &["bar", "baz"]).is_ok());
         assert_eq!(cl.as_str(), "foo=bar,baz");
     }
+
+    #[test]
+    fn test_partial_eq() {
+        let mut c1 = Cmdline::new(20);
+        let mut c2 = Cmdline::new(20);
```
Comment (lauralt) [inline src/cmdline/mod.rs:493]: Can we add a test for different max sizes, but the same string?

Comment (alexandruag) [reply src/cmdline/mod.rs]: Oh they were supposed to be different (i.e. 20 and 30), I've fixed this now.

**Review by lauralt — APPROVED** [review]

**Review by andreeaflorescu — APPROVED** [review]

**Pull request merged**